.github/workflows/build.yaml

```diff
@@ -258,13 +286,77 @@ jobs:
         if: matrix.config.apt_get != ''
         run: sudo apt-get update && sudo apt-get install ${{ matrix.config.apt_get }}
 
+      - name: Prepare Windows Environment
+        if: runner.os == 'Windows'
+        shell: powershell
+        run: |
+          $ErrorActionPreference = 'Stop'
+
+          # Determine VS2019 install path
+          $vsInstallPath = (Get-VSSetupInstance | `
+            Select-VSSetupInstance -Product '*' -Latest -Version '[16.0,17.0)' `
+          ).InstallationPath
+
+          $vcvarsall = "${vsInstallPath}\VC\Auxiliary\Build\vcvarsall.bat"
+          if (-not (Test-Path -Path $vcvarsall)) {
+            Out-Error "Could not find Visual Studio setup script"
+            exit 1
+          }
+
+          # Run VS environment script
+          $env:VSCMD_SKIP_SENDTELEMETRY = '1'
+          cmd /c "`"${vcvarsall}`" ${{ matrix.config.msvc_arch }} & set" | foreach {
+            if ($_ -match '^\w+=') {
+              $k, $v = $_.split('=', 2)
+              Set-Item -Force -Path "env:\${k}" -Value $v
+            }
+          }
+
+          # The test suite currently requires that the compiler specified by the "CC"
+          # env variable is on a path without spaces. Provide that by creating a
+          # junction from ~/opt/llvm to the Visual Studio path.
+          $null = New-Item `
+            -Path "${HOME}\opt\llvm" `
+            -ItemType Junction `
+            -Target "${vsInstallPath}\VC\Tools\Llvm\x64" `
+            -Force
+          $env:PATH = "$HOME\opt\llvm\bin;${env:PATH}"
+
+          # Export necessary VS environment variables
+          $msvcEnv = @(
+            "INCLUDE=${env:INCLUDE}",
+            "LIB=${env:LIB}",
+            "LIBPATH=${env:LIBPATH}",
+            "Path=${env:Path}",
+            "VCINSTALLDIR=${env:VCINSTALLDIR}",
+            "VCToolsInstallDir=${env:VCToolsInstallDir}",
+            "VSINSTALLDIR=${env:VSINSTALLDIR}"
+          )
+          $Utf8NoBomEncoding = New-Object System.Text.UTF8Encoding $False
+          [System.IO.File]::AppendAllLines(
+            $env:GITHUB_ENV,
+            [string[]]$msvcEnv,
+            $Utf8NoBomEncoding)
```

Comment:
Another stupid idea: does it make sense to use e.g. https://github.com/marketplace/actions/enable-developer-command-prompt? Sounds like it could simplify setup a lot.

Reply:
I'll have a look at the action and see how much of this can be replaced by it. Only disadvantage I can see is that it doesn't let you pick with Visual Studio major version to use, but it looks like the current GitHub runner only has 2019 installed so that's not really an issue.

Reply:
Looks a lot simpler now. I love the action marketplace 👍